dask/array/fft.py

```diff
@@ -24,21 +24,24 @@
     """
 
 
-def _fft_wrap(fft_func, dtype, out_chunk_fn):
-    def func(a, n=None, axis=-1):
-        if len(a.chunks[axis]) != 1:
-            raise ValueError(chunk_error % (axis, a.chunks[axis]))
+def _fft_wrap(dtype, out_chunk_fn):
```

Comment:
I do like your idea about detecting type better, @mrocklin. Let me give that some more thought.